ironfish/src/mining/director.ts

```diff
@@ -497,6 +498,34 @@ export class MiningDirector {
       clearTimeout(this.miningDifficultyChangeTimeout)
     }
   }
+
+  protected async *getTransactions(): AsyncGenerator<Transaction> {
+    const nullifiers = new BufferSet()
+
+    for (const transaction of this.memPool.get()) {
+      let conflicted = false
+
+      for (const spend of transaction.spends()) {
+        if (nullifiers.has(spend.nullifier)) {
+          conflicted = true
+          break
+        }
+
+        nullifiers.add(spend.nullifier)
+
+        if (await this.chain.nullifiers.contains(spend.nullifier)) {
+          conflicted = true
+          break
+        }
```

Comment:
There may be an edge case here, but the behavior might be intentional, I think we discussed this a bit yesterday.

Say you have two transactions in the queue, each with two nullifiers:
```
T1: N1, N2
T2: N1, N3
```
Assume N2 is in the chain already, but N1 and N3 are not. When the loop processes T1, N1 will get added to `nullifiers`, N2 will cause `conflicted` to be true and break out of the loop, then when the loop processes T2, N1 will be found to already be in `nullifiers` and break out of the loop, though it could potentially be added to the block.

Reply:
ah that's worth fixing, good find on my sloppy logic.

Reply:
Fixed